src/Razor/src/Microsoft.VisualStudio.Editor.Razor/TelemetryReporter.cs

```diff
@@ -145,6 +141,8 @@ private void Report(TelemetryEvent telemetryEvent)
             var name = telemetryEvent.Name;
             var propertyString = string.Join(",", telemetryEvent.Properties.Select(kvp => $"[ {kvp.Key}:{kvp.Value} ]"));
             _logger?.LogTrace("Telemetry Event: {name} \n Properties: {propertyString}\n", name, propertyString);
+
+            Debug.Assert(telemetryEvent.Name != GetTelemetryName("fault"), $"Fault Event: {name} \n Properties: {propertyString}");
```

Comment:
you should be able to do `telemetryEvent is not FaultEvent` for the condition

Reply:
Ahh, nice catch, thanks